tiledb/sm/fragment/fragment_metadata.h

```diff
@@ -787,6 +787,27 @@ class FragmentMetadata {
   Status load_fragment_min_max_sum_null_count(
       const EncryptionKey& encryption_key);
 
+  /**
+   * Checks if the fragment overlaps partially (not fully) with a given
+   * array open - end time.
+   *
+   * @param array_start_timestamps Array open time
+   * @param array_end_timestamp Array end time
+   *
+   * @return True if there is partial overlap, false if there is full or no
+   * overlap
+   */
+  inline bool partial_time_overlap(
+      uint64_t array_start_timestamp, uint64_t array_end_timestamp) const {
+    auto fragment_timestamp_start = timestamp_range_.first;
```

Comment:
NIT: const auto

Reply:
I also wonder if declaring the function parameters as const might allow some optimizations.

Reply:
const 🚓